apollo-router-core/src/services/mod.rs

```diff
@@ -52,6 +53,7 @@ impl TryFrom<ResponseBody> for Response {
             ResponseBody::RawString(_) => {
                 Err("wrong ResponseBody kind: expected Response, found RawString")
             }
+            ResponseBody::Text(_) => Err("wrong ResponseBody kind: expected Response, found Text"),
```

Comment:
NIT: we should probably define some static &str for these three strings to avoid the possibility of future diversion.